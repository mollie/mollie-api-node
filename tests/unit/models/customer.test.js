import Customer from 'models/customer';

describe('customer model', () => {
  it('should instantiate with default values', () => {
    const customer = new Customer();

    expect(customer.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const customerProps = JSON.parse(`{
        "resource": "customer",
        "id": "cst_kEn1PlbGa",
        "mode": "test",
        "name": "Customer A",
        "email": "customer@example.org",
        "locale": "nl_NL",
        "metadata": null,
        "recentlyUsedMethods": [
            "creditcard",
            "ideal"
        ],
        "createdAt": "2018-04-06T13:23:21.0Z",
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/customers/cst_kEn1PlbGa",
                "type": "application/hal+json"
            },
            "documentation": {
                "href": "https://docs.mollie.com/reference/v2/customers-api/get-customer",
                "type": "text/html"
            }
        }
    }`);
    const customer = new Customer(customerProps);

    expect(customer.toPlainObject()).toMatchSnapshot();
  });
});
