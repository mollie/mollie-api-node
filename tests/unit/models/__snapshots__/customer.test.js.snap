// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "mandates": null,
    "payments": null,
    "subscriptions": null,
  },
  "createdAt": null,
  "email": null,
  "id": null,
  "locale": null,
  "metadata": null,
  "mode": null,
  "name": null,
  "recentlyUsedMethods": null,
  "resource": "customer",
}
`;

exports[`customer model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/customers-api/get-customer",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/customers/cst_kEn1PlbGa",
      "type": "application/hal+json",
    },
  },
  "createdAt": "2018-04-06T13:23:21.0Z",
  "email": "customer@example.org",
  "id": "cst_kEn1PlbGa",
  "locale": "nl_NL",
  "metadata": null,
  "mode": "test",
  "name": "Customer A",
  "recentlyUsedMethods": Array [
    "creditcard",
    "ideal",
  ],
  "resource": "customer",
}
`;
